### ISSUE TITLE
aws_cloudwatch_event_target: diffs didn't match during apply

### ISSUE THREAD

**Issue by anosulchik** [hashicorp/terraform#5582]
Please refer #5581 -- I used a workaround to prevent terraform crash mentioned there [(please see this comment)](https://github.com/hashicorp/terraform/issues/5581#issuecomment-195322509). During re-creation of AWS lambda function (refer mentioned comment) terraform fails with "diffs didn't match during apply" error -- please see output below. After next run terraform reports "Apply complete! Resources: 0 added, 0 changed, 0 destroyed." but it still looks like a bug.

Error output of first TF apply run:

```
Error applying plan:

1 error(s) occurred:

* aws_cloudwatch_event_target.ecs_failed_deploys_lambda_target: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.

Please include the following information in your report:

    Terraform Version: 0.6.12
    Resource ID: aws_cloudwatch_event_target.ecs_failed_deploys_lambda_target
    Mismatch reason: attribute mismatch: arn
    Diff One (usually from plan): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff{"arn":*terraform.ResourceAttrDiff{Old:"arn:aws:lambda:us-east-1:000000000:function:tf-myservice-staging-ecs_failed_deploys", New:"${aws_lambda_function.ecs_failed_deploys_lambda.arn}", NewComputed:false, NewRemoved:false, NewExtra:interface {}(nil), RequiresNew:false, Type:0x0}}, Destroy:false, DestroyTainted:false}
    Diff Two (usually from apply): *terraform.InstanceDiff{Attributes:map[string]*terraform.ResourceAttrDiff(nil), Destroy:false, DestroyTainted:false}

Also include as much context as you can about your config, state, and the steps you performed to trigger this error.


Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

Resources definitions mentioned in error:

```
resource "aws_cloudwatch_event_target" "ecs_failed_deploys_lambda_target" {
  rule = "${aws_cloudwatch_event_rule.ecs_failed_deploys_lambda_schedule.name}"
  target_id = "InvokeLambda"
  arn = "${aws_lambda_function.ecs_failed_deploys_lambda.arn}"
}
```

```
resource "aws_lambda_function" "ecs_failed_deploys_lambda" {
  filename = "environments/${var.environment_name}/lambdas/ecs_failed_deploys.zip"
  function_name = "tf-${var.service_name}-${var.environment_name}-ecs_failed_deploys"
  role = "${aws_iam_role.iam_role.arn}"
  runtime = "python2.7"
  timeout = "8"
  handler = "lambda_function.lambda_handler"
}
```

Thank you!

**Comment by ElliotG** [comment]
+1, this is also happening for me. Currently a pretty obnoxious blocker, as every time we apply terraform to re-generate a lambda function we have to manually edit the state file.

**Comment by mitchellh** [comment]
Hello! This should be fixed in later versions of Terraform (0.7.11 at time of writing). 

Its hard to know 100% for a couple reasons:

  * This is on a now much-older version of Terraform. This is our fault, not yours, for not trying sooner, but we've fixed many diff mismatch issues since then.

  * We've fixed issues with exact error messages as this in those versions. 

Please try again! If the issue persists please just open a new issue. Thanks!
